src/components/tile-select/tile-select.test.tsx

```diff
@@ -0,0 +1,363 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import TileSelect, { TileSelectProps } from "./tile-select.component";
+import { testStyledSystemMargin } from "../../__spec_helper__/__internal__/test-utils";
+import TileSelectGroup, {
+  TileSelectGroupProps,
+} from "./tile-select-group/tile-select-group.component";
+import Button from "../button";
+import Icon from "../icon";
+
+testStyledSystemMargin((props) => <TileSelect {...props} />);
+
+testStyledSystemMargin((props) => (
+  <TileSelectGroup name="tile-select-group" {...props}>
+    <TileSelect name="test" />
+  </TileSelectGroup>
+));
+
+test("the deselect action button is rendered when TileSelect is checked", () => {
+  render(<TileSelect checked />);
+
+  expect(screen.getByRole("button")).toBeInTheDocument();
+});
+
+test("should disable deselect action button button if `disabled` prop is provided", () => {
+  render(<TileSelect checked disabled />);
+
+  expect(screen.getByRole("button")).toBeDisabled();
+});
+
+test("component should invoke passed onChange callback", async () => {
+  const user = userEvent.setup({ delay: null });
```

Comment:
**question:** Why is `delay: null` being set here and in other tests using `useEvent`? Not really familiar with it

Reply:
Messaged you privately regarding this 😄